### DISCUSSION TITLE
Custom sorting of directory items

### DISCUSSION THREAD

**Discussion by mishanja88** [gerbera/gerbera#3310] [Q&A]
Hello. I've decided to improve the sorting of my library items, but don't understand where to begin.
Could you please provide me any example of custom sorting JS scripts?

All I want is to order directory items in "windows-style", and not in standard "dictionary-style", because some folders contain, for example
```
1.first_item.avi
10.tenth_item.avi
11.eleventh_item.avi
2.second_item.avi
(and so on)
```
Human-oriented sorting would be, of course: 1,2,3... 9,10,11 (and so on)
I agree that the renaming or symlinking files would be the easiest solution, but I want to keep them as they are.
So my options are custom js scripting, or modifying gerbera source ( maybe, `fs::directory_iterator` override ).
Modifying source, though, will force me to maintain own custom branch.
I decided to stick to JS scripting option, but it's hard to find an example for my case.

Thank you for answering, in advance!

**Comment by udippel** [comment]
Yup. I think, this isn't very specific but could warrant filing an RFE, e.g. to be configurable.
